scipy/sparse/construct.py

```diff
@@ -664,16 +664,27 @@ def block_diag(mats, format=None, dtype=None):
            [0, 0, 0, 7]])
 
     """
-    nmat = len(mats)
-    rows = []
+    dtype = np.dtype(dtype)
```

Comment:
This line isn't necessary, as we're just passing the `dtype` through to other functions.

Reply:
thanks, I thought I had to explicitly handle `None` values for the `dtype` and `format`